### ISSUE TITLE
Inconsistent behaviour with `weight` argument in CrossEntropyLoss and BCEWithLogitsLoss

### ISSUE THREAD

**Issue by jamestszhim** [pytorch/pytorch#54843]
## 🐛 Bug

Inconsistent behaviour with `weight` argument in CrossEntropyLoss and BCEWithLogitsLoss

## To Reproduce

```
sm_target = torch.tensor([1,0,0])
b_target = torch.FloatTensor([1,0,0])

sm_score = torch.FloatTensor([[0.3, 0.7], [0.6, 0.4], [0.6, 0.4]])
b_score = torch.FloatTensor([0.40005136065843544, -0.19986264725851363, -0.19986264725851363])

print(F.cross_entropy(sm_score, sm_target))
print(F.binary_cross_entropy_with_logits(b_score, b_target))

# output
# tensor(0.5698)
# tensor(0.5698)

# setting class 0 with weight 1 and class 1 with weight 3
print(F.cross_entropy(sm_score, sm_target, weight = torch.FloatTensor([1, 3])))
print(F.binary_cross_entropy_with_logits(b_score, b_target, weight = torch.FloatTensor([3, 1, 1])))
# output
# tensor(0.5471)
# tensor(0.9118)
```
## Expected behavior

The loss from CrossEntropy for 2 classes and BCEWithLogitsLoss should be consistent with proper `weight` argument set.

## Environment

Should be environment-agnostic 

## Additional context

The issue should be related to an incorrect denominator used to compute the averaged loss.
```
def bce_1(target, score, weight):
    loss = 0.0
    for (y, yhat, w) in zip(target, score, weight):
        syhat = F.sigmoid(yhat)
        one_loss = w*(-y*torch.log(syhat)-(1-y)*torch.log(1-syhat))
        loss += one_loss.data
    return loss/len(target) # correct implementation should be sum(weight)#

print('BCE 1')
print(bce_1(b_target, b_score, weight=[1,1,1]))
print(bce_1(b_target, b_score, weight=[3,1,1])) # it doesn't include the weight

def bce_2(target, score, weight):
    loss = 0.0
    for (y, yhat, w) in zip(target, score, weight):
        syhat = F.sigmoid(yhat)
        one_loss = w*(-y*torch.log(syhat)-(1-y)*torch.log(1-syhat))
        loss += one_loss.data
    return loss/sum(weight)

print('BCE 2')
print(bce_2(b_target, b_score, weight=[1,1,1]))
print(bce_2(b_target, b_score, weight=[3,1,1])) # it doesn't include the weight

# output
# BCE 1
# tensor(0.5698)
# tensor(0.9118)
# BCE 2
# tensor(0.5698)
# tensor(0.5471) <- consistent with the loss in CrossEntropy

```
